tests/test_views.py

```diff
@@ -1,15 +1,17 @@
-import pytest
+from collections import OrderedDict
 
 
 def test_api_parse_succeeds(client):
-    # TODO: Finish this test. Send a request to the API and confirm that the
-    # data comes back in the appropriate format.
     address_string = '123 main st chicago il'
-    pytest.fail()
+    response = client.get('/api/parse?input=' + address_string, follow=True)
+    address_components = response.data['address_components']
+    address_type = response.data['address_type']
+    assert response.status_code == 200
+    assert isinstance(address_components, OrderedDict)
```

Comment:
why is it important to check if this is an ordereddict?

Reply:
This is a good question. I’m assuming you mean as opposed to a dict (but may be misunderstanding). I’m not actually sure — I was looking to test type, and I knew that usaddress.tag should return an OrderedDict, and so that’s what I’d expect my method to return. But as I think about it, there isn’t necessarily any reason it would need to be an OrderedDict in this particular instance, given that dict, for the version of Python in this project, would also maintain order, and also given the fact that I’m not testing usaddress itself anyway, just making sure my API endpoint returns the expected output for address_components and address_type. But I’d be really curious to hear your thoughts.